go/batch-submitter/dial_l2_client.go

```diff
@@ -1,23 +1,24 @@
-package dial
+package batchsubmitter
 
 import (
 	"context"
 	"crypto/tls"
 	"net/http"
 	"strings"
 
+	"github.com/ethereum-optimism/optimism/go/bss-core/dial"
 	"github.com/ethereum-optimism/optimism/l2geth/ethclient"
 	"github.com/ethereum-optimism/optimism/l2geth/log"
 	"github.com/ethereum-optimism/optimism/l2geth/rpc"
 )
 
-// L2EthClientWithTimeout attempts to dial the L2 provider using the
-// provided URL. If the dial doesn't complete within defaultDialTimeout seconds,
+// DialL2EthClientWithTimeout attempts to dial the L2 provider using the
+// provided URL. If the dial doesn't complete within dial.DefaultTimeout seconds,
 // this method will return an error.
-func L2EthClientWithTimeout(ctx context.Context, url string, disableHTTP2 bool) (
+func DialL2EthClientWithTimeout(ctx context.Context, url string, disableHTTP2 bool) (
```

Comment:
Is there any reason as to why we are using the `l2geth` deps here? I feel like we could use the upstream stuff for things like rpc or utils, especially now that the http2 fix is in: https://github.com/ethereum-optimism/optimism/commit/6f2ea193f432a597a83709327dd510319b27a03e

Reply:
yes, upstream doesn't expose the L2 metadata https://github.com/ethereum-optimism/optimism/blob/develop/go/batch-submitter/drivers/sequencer/batch.go#L53L54

Reply:
fwiw we do use upstream for L2 in this driver since we don't need to access any non-standard info